internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -49,7 +51,9 @@ func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AServ
 		return nil, err
 	}
 
-	// TODO(rmehta19): Handle resp.GetStatus().
+	if resp.GetStatus().Code != uint32(codes.OK) {
```

Comment:
Sorry I just noticed this: can we hit this if statement in a unit test?

If this PR is blocking other PRs in the queue, I'm fine adding a TODO for now and doing this in a follow-up.

Similarly below.

Reply:
Will do! 

I think this will involve catching an error somewhere else on the fakes2av2 side. Currently fakes2av2 only populates the status with a non OK status when the connection side is invalid, the connection side is something that get's populated in the GetTlsConfigurationForClient/Server. I have a unit test in fakes2av2_test that tests this  already, but it directly makes the RPC with an invalid connection side rather than calling GetTlsConfigurationForClient/Server which makes the RPC.

For now, I can add to fakes2av2: read the SNI in GetTlsConfigurationReq, if SNI == "throw_internal_error", then throw an internal error. This will get us code coverage for this line, does that sound ok? I will add a TODO to get to this in another PR, since it involves more files changed than just these.

Reply:
SGTM! Thanks!